requests/models.py

```diff
@@ -924,6 +924,18 @@ def raise_for_status(self):
                 reason = self.reason.decode('iso-8859-1')
         else:
             reason = self.reason
+        
+        if isinstance(self.text, bytes):
```

Comment:
This doesn't do anything sensible. `self.text` is definitionally a unicode string, so this block will never be entered. We can safely skip it.

*However*, this is also not necessarily a safe thing to do: see my more general feedback.